DQMServices/Core/interface/LegacyIOHelper.h

```diff
@@ -16,8 +26,14 @@ class LegacyIOHelper {
   // use internal type here since we call this from the DQMStore itself.
   typedef dqm::implementation::DQMStore DQMStore;
   typedef dqm::legacy::MonitorElement MonitorElement;
-  LegacyIOHelper(DQMStore* dqmstore) : dbe_(dqmstore){};
 
+  typedef dqm::harvesting::DQMStore HDQMStore;
+  typedef dqm::harvesting::MonitorElement HMonitorElement;
+
+  using MEMap = std::set<HMonitorElement*>;
+
+  LegacyIOHelper(DQMStore* dqmstore) : dbe_(dqmstore){};
+  LegacyIOHelper(HDQMStore* hdqmstore) : dbe_(hdqmstore){};
```

Comment:
Thanks for this functionality @sroychow 
I am not sure why you need here a HarvestedDQMStore.

At present, dqm::harvesting and dqm::legacy are identical, but dqm::reco provides less operations on the MonitorElement. The common implementation is in the internal namespace dqm::implementation. There are inheritance relations between the different variations to allow implicit conversions in some common cases.

In any case, to avoid misunderstanding with HDQM (Historical DQM), may I suggest another name like:
typedef dqm::harvesting::DQMStore HDQMStore; ->
typedef dqm::harvesting::DQMStore HarvestedDQMStore;   or
typedef dqm::harvesting::DQMStore LegacyDQMStore;

hence:
LegacyIOHelper(HDQMStore* hdqmstore) : dbe_(hdqmstore){}; ->
LegacyIOHelper(HDQMStore* harvesteddqmstore) : dbe_(harvesteddqmstore){};

Reply:
>  I am not sure why you need here a HarvestedDQMStore.

we need the harvesting flavour since the `SiStripOfflineDQM` plugin is using harvesting flavour of ME and DQMStore.

> In any case, to avoid misunderstanding with HDQM (Historical DQM), may I suggest another name:

that's implemented in the last push.